File: priv/combined_metrics/samples/variable_naming/name_is_abbreviation/bad/api_client.rb
# HTTP API client using abbreviated variable names.
# BAD: variables like usr, cfg, req, res, addr, msg obscure intent.

require 'net/http'
require 'json'
require 'uri'

class ApiClientBad
  def send_request(cfg, usr)
    req = build_req(cfg, usr)
    addr = URI("#{cfg[:base_url]}#{req[:path]}")

    res = Net::HTTP.post(addr, req[:body], req[:headers])
    msg = JSON.parse(res.body)

    { ok: res.is_a?(Net::HTTPSuccess), data: msg }
  rescue => err
    { ok: false, error: "Request failed: #{err.message}" }
  end

  def fetch_product(cfg, prd_id)
    addr = URI("#{cfg[:base_url]}/products/#{prd_id}")
    req = Net::HTTP::Get.new(addr)
    req['Authorization'] = "Bearer #{cfg[:api_key]}"

    res = Net::HTTP.start(addr.host, addr.port) { |http| http.request(req) }

    unless res.is_a?(Net::HTTPSuccess)
      msg = "Unexpected status: #{res.code}"
      return { ok: false, error: msg }
    end

    prd = JSON.parse(res.body)
    { ok: true, data: prd }
  rescue => err
    { ok: false, error: err.message }
  end

  def create_order(cfg, usr, qty)
    addr = URI("#{cfg[:base_url]}/orders")
    req = Net::HTTP::Post.new(addr)
    req['Authorization'] = "Bearer #{cfg[:api_key]}"
    req['Content-Type'] = 'application/json'
    req.body = JSON.dump({ user_id: usr[:id], quantity: qty })

    res = Net::HTTP.start(addr.host, addr.port) { |http| http.request(req) }
    msg = JSON.parse(res.body)

    return { ok: false, error: extract_err_msg(msg) } unless res.is_a?(Net::HTTPSuccess)

    { ok: true, data: msg }
  rescue => err
    { ok: false, error: err.message }
  end

  def paginate(cfg, params)
    qry = URI.encode_www_form(params)
    addr = URI("#{cfg[:base_url]}/items?#{qry}")
    req = Net::HTTP::Get.new(addr)
    req['Authorization'] = "Bearer #{cfg[:api_key]}"

    res = Net::HTTP.start(addr.host, addr.port) { |http| http.request(req) }
    msg = JSON.parse(res.body)

    { ok: res.is_a?(Net::HTTPSuccess), data: msg }
  rescue => err
    { ok: false, error: err.message }
  end

  def upload_file(cfg, usr, buf)
    addr = URI("#{cfg[:base_url]}/uploads")
    req = Net::HTTP::Post.new(addr)
    req['Authorization'] = "Bearer #{cfg[:api_key]}"
    req.body = buf
    req['X-User-Id'] = usr[:id].to_s

    res = Net::HTTP.start(addr.host, addr.port) { |http| http.request(req) }
    msg = JSON.parse(res.body)

    { ok: res.is_a?(Net::HTTPSuccess), data: msg }
  end

  private

  def build_req(cfg, usr)
    { path: '/requests', body: JSON.dump({ user_id: usr[:id] }), headers: auth_headers(cfg) }
  end

  def auth_headers(cfg) = { 'Authorization' => "Bearer #{cfg[:api_key]}" }
  def extract_err_msg(msg) = msg['error'] || 'Unknown error'
end

File: priv/combined_metrics/samples/variable_naming/name_is_abbreviation/bad/api_client.js
// HTTP API client using abbreviated variable names.
// BAD: variables like usr, cfg, req, res, addr, msg obscure intent.

async function sendRequest(cfg, usr) {
  const req = buildReq(cfg, usr);
  const addr = cfg.baseUrl + req.path;

  try {
    const res = await fetch(addr, {
      method: req.method,
      headers: req.headers,
      body: req.body,
    });

    const msg = await res.json();
    return { ok: true, data: msg };
  } catch (err) {
    return { ok: false, error: `Request failed: ${err.message}` };
  }
}

async function fetchProduct(cfg, prdId) {
  const addr = `${cfg.baseUrl}/products/${prdId}`;
  const req = { headers: authHeaders(cfg) };

  try {
    const res = await fetch(addr, req);

    if (!res.ok) {
      const msg = `Unexpected status: ${res.status}`;
      return { ok: false, error: msg };
    }

    const prd = await res.json();
    return { ok: true, data: prd };
  } catch (err) {
    return { ok: false, error: err.message };
  }
}

async function createOrder(cfg, usr, qty) {
  const addr = `${cfg.baseUrl}/orders`;
  const req = {
    method: 'POST',
    headers: { ...authHeaders(cfg), 'Content-Type': 'application/json' },
    body: JSON.stringify({ userId: usr.id, quantity: qty }),
  };

  try {
    const res = await fetch(addr, req);
    const msg = await res.json();

    if (!res.ok) return { ok: false, error: extractErrMsg(msg) };

    return { ok: true, data: msg };
  } catch (err) {
    return { ok: false, error: err.message };
  }
}

async function paginate(cfg, params) {
  const qry = new URLSearchParams(params).toString();
  const addr = `${cfg.baseUrl}/items?${qry}`;
  const req = { headers: authHeaders(cfg) };

  try {
    const res = await fetch(addr, req);
    const msg = await res.json();
    return { ok: true, data: msg };
  } catch (err) {
    return { ok: false, error: err.message };
  }
}

async function uploadFile(cfg, usr, buf) {
  const addr = `${cfg.baseUrl}/uploads`;
  const frm = new FormData();
  frm.append('file', buf);
  frm.append('userId', usr.id);

  const res = await fetch(addr, { method: 'POST', headers: authHeaders(cfg), body: frm });
  const msg = await res.json();
  return { ok: res.ok, data: msg };
}

function buildReq(cfg, usr) {
  return { method: 'POST', path: '/requests', body: JSON.stringify({ userId: usr.id }), headers: authHeaders(cfg) };
}

function authHeaders(cfg) {
  return { Authorization: `Bearer ${cfg.apiKey}` };
}

function extractErrMsg(msg) {
  return msg.error || 'Unknown error';
}

module.exports = { sendRequest, fetchProduct, createOrder, paginate, uploadFile };
